### ISSUE TITLE
Is it possible to pause video from timeupdate handler while seeking?

### ISSUE THREAD

**Issue by jpfranco** [videojs/video.js#2226]
So imagine that you need to pause the player and show an overlay when a video reaches a certain time or when the user seeks past a certain time. The first scenario is easy enough, simply listen to `timeupdate` and call `player.pause()` when you reach the desired time. When the video is playing and the user seeks past the threshold, however, it seems like it is not possible to reliably stop the player. It appears that `play` is called once again when seeking ends, but I'm not really sure why.

If you take a look at [this example](http://jsfiddle.net/z1vqrs8y/8/) and you seek past 5 seconds, you'll see that the video is restarted after pausing it from within `timeupdate`. I'm trying to figure out whether this is a bug in VideoJS or just something that one shouldn't be doing inside the `timeupdate` handler. Pausing the video in this manner seems to work ok in Safari, but not in Chrome and Firefox. Thanks in advance.

**Comment by heff** [comment]
The player pauses while scrubbing so if you grab the handle and scrub to a
place and stop, it doesn't keep playing. I'm not totally clear on what
you're trying to do, but does that help explain part of it?

**Comment by jpfranco** [comment]
Sorry, I think I didn't explain it properly. I'm trying to achieve something similar to what Hulu does with ads. You have ad breaks configured at certain points during the video, and if the user tries to scrub past an ad break, the player stops playback, displays the ad and then resumes content playback after the ad is complete.

You're right that when the video is already paused, the player will remain paused after the scrubbing is complete, but if the video was playing when the scrubbing began, then playback will resume after the player is done seeking. This is ok, except that if you happen to programmatically pause the video while it is seeking (i.e. by calling `pause()` from within a `timeupdate` handler that was triggered by the scrubbing action), then it seems that the video will always resume after the scrubbing is complete.

To summarize, my problem is that I'm not able to programmatically prevent a user from scrubbing past a certain time because the player always resumes after seeking is complete (provided that the video wasn't initially paused). This is a bit hard to explain, I hope that this wasn't even more confusing.

**Comment by OwenEdwards** [comment]
I've looked at something like this, and you'll need to track the `seeking` and `seeked` events. Once seeking starts, save the current time, and disable the `timeupdate` handler. Then, on seeked, re-enable the `timeupdate` handler, and if the current time has crossed the boundary (in your example, 5 seconds), pause the video, and play your ad.

My understanding is that `timeupdate` events are still fired during scrubbing to allow the UI to update (including captions), but those events don't make sense for certain things.

**Comment by jpfranco** [comment]
Thanks @OwenEdwards, this is what I needed to know. I guess it makes more sense to approach this issue with the `seeked` events, I'll give that a try.
